mods/cnc/maps/nod05/nod05.lua

```diff
@@ -0,0 +1,317 @@
+	end)
+
+	Trigger.OnObjectiveCompleted(Nod, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+
+	Trigger.OnObjectiveFailed(Nod, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerWon(Nod, function()
+		Media.PlaySpeechNotification(Nod, "Win")
+	end)
+
+	Trigger.OnPlayerLost(Nod, function()
+		Media.PlaySpeechNotification(Nod, "Lose")
+	end)
+
+	NodObjective1 = Nod.AddPrimaryObjective("Build 3 SAMs")
+	NodObjective2 = Nod.AddPrimaryObjective("Destroy the GDI base")
+	GDIObjective = GDI.AddPrimaryObjective("Kill all enemies!")
+
+	AirstrikeProxy = Actor.Create("airstrike.proxy", true, { Owner = GDI })
+
+	Trigger.AfterDelay(XxxxTriggerFunctionTime, XxxxTriggerFunction)
+	Trigger.AfterDelay(YyyyTriggerFunctionTime, YyyyTriggerFunction)
+	Trigger.AfterDelay(ZzzzTriggerFunctionTime, ZzzzTriggerFunction)
+
+	Trigger.OnEnteredFootprint(DelyCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			YyyyTriggerSwitch = true
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnEnteredFootprint(DelzCellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			ZzzzTriggerSwitch = true
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.AfterDelay(Grd1TriggerFunctionTime, Grd1TriggerFunction)
+
+	Trigger.OnEnteredFootprint(Atk5CellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			Atk5TriggerFunction()
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.AfterDelay(Atk2TriggerFunctionTime, Atk2TriggerFunction)
+	Trigger.AfterDelay(Atk3TriggerFunctionTime, Atk3TriggerFunction)
+	Trigger.AfterDelay(Atk4TriggerFunctionTime, Atk4TriggerFunction)
+	Trigger.AfterDelay(Atk6TriggerFunctionTime, Atk6TriggerFunction)
+
+	Trigger.OnEnteredFootprint(Atk1CellTriggerActivator, function(a, id)
+		if a.Owner == Nod then
+			Atk1TriggerFunction()
+			Trigger.RemoveFootprintTrigger(id)
+		end
+	end)
+
+	Trigger.OnDiscovered(Tower1, AutoTriggerFunction)
+	Trigger.OnDiscovered(Tower2, AutoTriggerFunction)
+
+	Trigger.OnAllRemovedFromWorld(HuntActorTriggerActivator, HuntTriggerFunction)
+
+	Trigger.AfterDelay(0, getStartUnits)
+end
+
+Tick = function()
+	if Nod.HasNoRequiredUnits() then
+		if DateTime.GameTime > 2 then
+			GDI.MarkCompletedObjective(GDIObjective)
+		end
+	end
+
+	if not Nod.IsObjectiveCompleted(NodObjective1) and CheckForSams(Nod) then
+		Nod.MarkCompletedObjective(NodObjective1)
+	end
+
+	if GDI.HasNoRequiredUnits() then
+		Nod.MarkCompletedObjective(NodObjective2)
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(3) == 0 then
+		checkProduction(GDI)
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(45) == 0 then
+		AutoTriggerFunction()
+	end
+end
+
+IdleHunt = function(unit)
+	if not unit.IsDead then
+		Trigger.OnIdle(unit, unit.Hunt)
+	end
+end
+
+CheckForSams = function(player)
+	local checked = { }
```

Comment:
Actually you're not using this, and I also don't think you need it here.
So what about just:
```lua
CheckForSams = function(player)
	local baseBuildings = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
		return actor.Owner == Nod and actor.Type == 'sam'
 	end)
	return #baseBuildings >= 3
end